runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/EngineRule.java

```diff
@@ -221,14 +213,14 @@ public Statement apply(
                 if (configurationRoot != null)
                 {
                     String resourceName = String.format("%s/%s", configurationRoot, config.value());
-
-                    configURL = testClass.getClassLoader().getResource(resourceName);
+                    String configURL = requireNonNull(testClass.getClassLoader().getResource(resourceName)).toString();
+                    configure(ENGINE_CONFIG_URL.name(), configURL);
                 }
                 else
                 {
                     String resourceName = String.format("%s-%s", testClass.getSimpleName(), config.value());
-
-                    configURL = testClass.getResource(resourceName);
+                    String configURL = requireNonNull(testClass.getClassLoader().getResource(resourceName)).toString();
+                    configure(ENGINE_CONFIG_URL.name(), configURL);
```

Comment:
This one should be using `testClass.getResource(...)` instead of `testClass.getClassLoader().getResource(...)`.

Note: there are quite a few calls to `configURL.toString()` in the codebase, consider changing the type of the `ENGINE_CONFIG_URL` property to `URL` instead of `String` to eliminate them.